orttraining/orttraining/training_ops/cuda/math/softmax_grad_impl.cu

```diff
@@ -133,7 +236,7 @@ template <typename T>
 Status SoftmaxGradImpl(cudaStream_t stream, cudnnHandle_t cudnn_handle, T* input_grad, const T* output_grad,
```

Comment:
In the PR description, we have batch_count > 8192, and see gains, do you think that is also true for batch_count < 8192. Maybe we should do  a quick benchmark for a few numbers to prove it?